### PR TITLE
Fixed Sales Reports for Sales transaction summary, Sales target variance, Sales order trends, Territory target variance report.

### LINKED ISSUE #995: Report of Sales Targets for a Group
At present a Sales target is set only for a Sales person and the report for the same i.e. - "Sales Person Target Variance (Item Group-Wise)"  - Under Selling works as expected.

But if :
a)  Sales Person is created with a - "Has Child Node" as Yes to enable a Team or a Group 
b) the Sales target is set for that "Group Sales Person" 
c) the Sales Persons under this "Group Sales Person" does a sale. 
Then
The Report -  "Sales Person Target Variance (Item Group-Wise)" shows only "0" values and not the actual figures.

If the targets for the group and individual sales person both work and proper reporting of the same is present in this  "Sales Person Target Variance (Item Group-Wise)"  report it will be great.

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/1040075-report-of-sales-targets-for-a-group?utm_campaign=plugin&utm_content=tracker%2F266088&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F266088&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #4046: Reporting error on "Sales Person Target Variance Item Group" Report
The "Total Achieved" values in Sales Person Target Variance Item Group report do not seem to be based on the Sales Person Contribution % on the order.

Attached:

1) Report-view.png: Highlighted in red shows the total order value, not the sales person contribution. 
2) Order-view.png: Shows the order total and sales person contribution amount.

I expect the "Total Achieved" values in the report to = sales person contribution on the order, not the order total.

Is that the correct expectation?

Thank you.
report-view
order-vew

### PR DESCRIPTION
**Pull request by shreyasp**
Fixed Territory target variance report.

Fixed #4046 
Fixed #995

### CODE REVIEW

File: erpnext/selling/report/sales_person_target_variance_item_group_wise/sales_person_target_variance_item_group_wise.py [hunk 1]
```diff
@@ -61,39 +61,66 @@ def get_columns(filters):
...
+			sales_person, item_code, month_name
+			""",
+		(start_date, end_date, lft, rgt), as_dict=1)
 
 	item_actual_details = {}
 	for d in item_details:
-		item_actual_details.setdefault(d.sales_person, {}).setdefault(\
-			get_item_group(d.item_code), []).append(d)
+		item_group = item_groups[d.item_code]
+		print item_group
```
Comment (anandpdoshi) [inline erpnext/selling/report/sales_person_target_variance_item_group_wise/sales_person_target_variance_item_group_wise.py]: Yo! you left a `print` :)

File: erpnext/selling/report/sales_person_wise_transaction_summary/sales_person_wise_transaction_summary.py [hunk 2]
```diff
@@ -35,34 +45,38 @@ def get_columns(filters):
...
 			conditions.append("dt.{0}=%s".format(field))
 			values.append(filters[field])
-			
+
 	if filters.get("sales_person"):
-		conditions.append("st.sales_person=%s")
-		values.append(filters["sales_person"])
-		
+		lft, rgt = frappe.get_value("Sales Person", filters.get("sales_person"), ["lft", "rgt"])
+		conditions.append("exists(select name from `tabSales Person` where lft >= %s and rgt <= %s and name=st.sales_person)" % (lft, rgt))
```
Comment (anandpdoshi) [inline erpnext/selling/report/sales_person_wise_transaction_summary/sales_person_wise_transaction_summary.py]: use `.format(lft, rgt)` next time.

File: erpnext/selling/sales_common.js [hunk 3]
```diff
@@ -82,10 +82,6 @@ erpnext.selling.SellingController = erpnext.TransactionController.extend({
 				}
 			});
 		}
-
-		if(this.frm.fields_dict.sales_team && this.frm.fields_dict.sales_team.grid.get_field("sales_person")) {
-			this.frm.set_query("sales_person", "sales_team", erpnext.queries.not_a_group_filter);
-		}
```
Comment (anandpdoshi) [inline erpnext/selling/sales_common.js]: why did you remove this?

Comment (shreyasp) [inline erpnext/selling/sales_common.js]: At present, we do not allow Parent to be selected in the sales-person field as part of the team. This happens since not_a_group_filter() filters out the parent node and Hence, we have removed the code to avoid filtering the parent node.

File: erpnext/selling/report/sales_person_wise_transaction_summary/sales_person_wise_transaction_summary.py (hunk 2 above)
Comment (shreyasp) [inline erpnext/selling/report/sales_person_wise_transaction_summary/sales_person_wise_transaction_summary.py]: Updated.

File: erpnext/selling/report/sales_person_wise_transaction_summary/sales_person_wise_transaction_summary.py [hunk 4]
```diff
@@ -12,13 +13,22 @@ def execute(filters=None):
...
 		data.append([
 			d.name, d.customer, d.territory, d.posting_date, d.item_code,
 			item_details.get(d.item_code, {}).get("item_group"), item_details.get(d.item_code, {}).get("brand"),
 			d.qty, d.base_net_amount, d.sales_person, d.allocated_percentage, d.contribution_amt
 		])
 
+	if data:
+		total_row = [""]*len(data[0])
+		total_row[0] = _("Total")
+		total_row[-1] = total_contribution_amoount
```
Comment (anandpdoshi) [inline erpnext/selling/report/sales_person_wise_transaction_summary/sales_person_wise_transaction_summary.py]: typo: `amoount`

**Comment by anandpdoshi** [comment]
will merge manually
